qiling/core.py

```diff
@@ -267,6 +275,20 @@ def shellcode(self):
         loader_shellcode(self)
 
     def run(self):
+
+        # setup syscall filter for logger
+        if self.syscall_filter != None and self.output == QL_OUT_DEFAULT:
```

Comment:
we need to move this to somewhere more proper, core.py try to me as os/arch friendly as possible. syscall is too *nix